src/control.rs

```diff
@@ -103,9 +103,16 @@ impl ShouldColorize {
     /// `CLICOLOR_FORCE` takes highest priority, followed by `NO_COLOR`,
     /// followed by `CLICOLOR` combined with tty check.
     pub fn from_env() -> Self {
+        #[allow(unused_mut)]
+        #[allow(unused_assignments)]
+        let mut tty = false;
+        #[cfg(feature = "tty")]
+        {
+            tty = atty::is(atty::Stream::Stdout);
+        }
```

Comment:
```suggestion
        let tty = if cfg!(feature = "tty") {
            atty::is(atty::Stream::Stdout)
        } else {
            false
        };
```

Reply:
Good idea. I like this better than mine but it doesn't compile. This compiles:

```
        let tty = if cfg!(feature = "tty") {
            #[cfg(feature = "tty")]
            {
            atty::is(atty::Stream::Stdout)
            }
            #[cfg(not(feature = "tty"))]
            false
        } else {
            false
        };
```

But we can clean it up further to this:

```
        let tty = {
            #[cfg(feature = "tty")]
            {
                atty::is(atty::Stream::Stdout)
            }
            #[cfg(not(feature = "tty"))]
            false
        };
```

And sorry there are some commits that I didn't mean to put into this PR.

Reply:
You could also use

```rust
use std::io::{self, IsTerminal};
let tty = io::stdout().is_terminal();
```

that was very recently [introduced in the Rust 1.70.0 std library](https://doc.rust-lang.org/stable/std/io/trait.IsTerminal.html).

Reply:
Could use that and drop the `atty` dependency.